Reply/app/src/main/java/com/example/reply/ui/ReplyHomeViewModel.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.reply.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.reply.data.Email
+import com.example.reply.data.EmailsRepository
+import com.example.reply.data.EmailsRepositoryImpl
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class ReplyHomeViewModel(private val emailsRepository: EmailsRepository = EmailsRepositoryImpl()): ViewModel() {
+
+    // UI state exposed to the UI
+    private val _uiState = MutableStateFlow(ReplyHomeUIState(loading = true))
+    val uiState: StateFlow<ReplyHomeUIState> = _uiState
+
+    init {
+        observeEmails()
+    }
+
+    private fun observeEmails() {
+        _uiState.value = ReplyHomeUIState(loading = true)
+        viewModelScope.launch {
+            emailsRepository.getAllEmails()
+                .catch { ex ->
+                    _uiState.value = ReplyHomeUIState(error = ex.message)
+                }
+                .collect { emails ->
+                _uiState.value = ReplyHomeUIState(emails = emails)
```

Comment:
The formatting here is a bit strange, before this gets merged into main we'll want to make sure we're applying linting and have a workflow for this project.